RollGen.Tests.Unit/ExplodeTests.cs

```diff
@@ -33,7 +35,7 @@ public class ExplodeTests : IntegrationTests
         public int ExplodeTest(int quantity, int die, int[] rolls)
         {
             var random = new MockRandom(rolls);
-            var partialRoll = new NumericPartialRoll(quantity, random, GetNewInstanceOf<ExpressionEvaluator>());
+            var partialRoll = new NumericPartialRoll(quantity, random, evaluator);
```

Comment:
Is there a reason you made your `MockRandom` class, instead of using `Mock<Random>`?  Also, you should be using a mock of the `ExpressionEvaluator` as well

Reply:
I wanted to be able to provide an array to return in order, as far as I could tell `Mock<Random>` would only return A single value, did I miss something you're aware of?

As for the expression evaluator, we want to be using albatross, we're not mocking it because we want it to work as it would normally.

Reply:
With `Moq`, you can set up a return sequence that will emulate your array.  Something like

```
mockRandom.SetupSequence(r => r.Next())
    .Returns(1)
    .Returns(0)
    .Returns(9000);
```

As for the expression evaluator, in reality we're not testing the albatross engine here, just our own explode logic.  However, setting up the mock expression evaluator would be somewhat involved, so I guess it would be easier to just use the real thing in this case.  If I feel the need, I'll refactor it later.

Reply:
Awesome, I pushed a fix to use a mocked return sequence.

Yeah, that was the way I felt about the evaluator, ideally we could mock it but we need good responses... I suppose we could, for each- Wait, we don't even use the evaluator for this, because we build the expression and we can parse it on our own, I can just mock it!
I'll push that in a bit.

Reply:
There we go. And I was thinking we could pass into the tests, through test parameters, what it should be after evaluation, should we need the evaluator to be mocked for real.